homeworks/03-data-structure/src/BinaryTree.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  breadthTraverse,
+  inorderTraverse,
+  postorderTraverse,
+  preorderTraverse,
+  TraverseType,
+} from "./Traverse"
+
+export class TreeNode<T> {
+  left?: TreeNode<T>
+  right?: TreeNode<T>
+  constructor(readonly value: T) {}
+}
+
+export interface BinaryTreeInterface<T> {
+  setTree(tree: TreeNode<T>): this
```

Comment:
`constructor` is absent.

Reply:
@shpakkdv
I have a problem with it. Could you take a look please?

When I add constructor in body of interface:
```constructor(tree?: TreeNode<T>): void```

I see a typescript error
![Screenshot 2021-05-03 at 12 07 37](https://user-images.githubusercontent.com/20773518/116859291-6dfdd680-ac08-11eb-8468-2ea972f310b4.png)

This requires a ```Function``` declaration, which is almost useless because it doesn't describe the types of the arguments.

How would you handle it?

Reply:
Sorry for misleading you. Nowadays, still no good way to define the `constructor` type. Never mind this comment.